neo/Prompt/Commands/Config.py

```diff
@@ -196,11 +199,15 @@ def execute(self, arguments):
 
             current_max = settings.CONNECTED_PEER_MAX
             connected_count = len(nodemgr.nodes)
-            if c1 < current_max and c1 < connected_count:
+            if current_max < connected_count:
                 to_remove = connected_count - c1
+                r_list = []
+                i = 1
                 for _ in range(to_remove):
-                    last_connected_node = nodemgr.nodes[-1]
-                    wait_for(last_connected_node.disconnect())  # need to avoid it being labelled as dead/bad
+                    r_list.append(nodemgr.nodes[-i])
+                    i += 1
```

Comment:
You previously said:
```
This additional list (r_list) is required because iterating with nodemgr.nodes[-1] called the same node again and again since nodemgr.nodes wasn't updated immediately. I added several print statements for debugging and tested manually in privatenet, which yielded this example:

Minpeers set to 1
Connected Count:  3
To Remove:  2
Removed 127.0.0.1:20335
Removed 127.0.0.1:20335
Maxpeers set to 1
The new method builds the list of nodes to remove first, then removes each node on the list.
```
Can you explain what's wrong with that output?

When I only apply this fix:
```
if current_max < connected_count:
```

and insert a print statement as follows
```
                for _ in range(to_remove):
                    last_connected_node = nodemgr.nodes[-1]
                    print(f"disconnecting {last_connected_node.address}")
                    wait_for(last_connected_node.disconnect())
```

then I get
```
neo> show nodes
Connected: 4 of max 5
Peer 0  /NEO:2.8.0/       127.0.0.1:20335 height:     2016
Peer 1  /NEO:2.8.0/       127.0.0.1:20333 height:     2016
Peer 2  /NEO:2.8.0/       127.0.0.1:20334 height:     2016
Peer 3  /NEO:2.8.0/       127.0.0.1:20336 height:     2016

neo> config maxpeers 1
disconnecting 127.0.0.1:20336
disconnecting 127.0.0.1:20334
disconnecting 127.0.0.1:20333
Maxpeers set to 1
neo> show nodes
Connected: 1 of max 1
Peer 0  /NEO:2.8.0/       127.0.0.1:20335 height:     2016

neo>
```

That seems to do exactly what it needs to do. Disconnect the last 3 nodes and only leave the first one that was connected.

Reply:
I will check if the behavior has improved in the latest commit, but you can see that the original behavior tried to remove the same node twice so you were stuck with connected peers > maxpeers.

Reply:
Okay, I just checked and it seems to be working the way you previously describe. I think maybe `wait_for` has improved?

Reply:
`wait_for` has not changed. It is just a utility function that allows run asynchronous code in a blocking fashion. Therefore I couldn't see how it would pick the same node twice because it waits for the node to be disconnected before remove the next one. Did you perhaps run it with your `run_coroutine_threadsafe` instead? because that could cause it as that creates tasks.

Reply:
Ahh okay, that might have been the issue. I had originally chosen not to use `wait_for` because of the previously mentioned Runtime error, but it seems it may have caused unintended consequences.